src/effective-dart/usage.md

```diff
@@ -350,13 +350,18 @@ then it probably does make sense to have a separate boolean field.
 
 Checking that a nullable variable is not equal to `null` promotes the variable
 to a non-nullable type. That lets you access members on the variable and pass it
-to functions expecting a non-nullable type. Unfortunately, promotion is only
-sound for local variables and parameters, so fields and top-level variables
-aren't promoted.
+to functions expecting a non-nullable type.
 
-One pattern to work around this is to assign the field's value to a local
-variable. Null checks on that variable do promote, so you can safely treat
-it as non-nullable.
+Type promotion is only sound, however, for local variables, parameters, and
+private final fields. Instances that are open to manipulation, whether
+explicitly, like top-level variables, or implicitly, like fields with a concrete
+getter of the same name in the same library, cannot be type promoted.
+Read [Fixing type promotion failures][] for a full list of non-promotion reasons.
+
+Declaring members [private] and [final], as we generally recommend, bypasses
```

Comment:
Technically, declaring members private and final isn't sufficient to bypass a conflict with a getter. Maybe change "bypasses these limitations" to "can often bypass these limitations"?